docs/integrations/gitlab.md

```diff
@@ -0,0 +1,74 @@
+| [Org](/pipes/docs/accounts/org) | [Team](/pipes/docs/accounts/org#team-plan) or [Enterprise](/pipes/docs/accounts/tenant#enterprise-plan)  | Mods are available to all workspaces in the organization.
+| [User (Developer Account)](/pipes/docs/accounts/developer) | any | Mods are available to all workspaces in the account.
+
+First, navigate to the **Integrations** page for the appropriate resource:
+- To configure a GitLab integration for your **Developer Account**, click the double arrow button from the navigation at the top of the page, select your account from the dropdown, and then select the **Integrations** tab to manage integrations for your account.
+- To configure a GitLab integration for your **Organization**, click the double arrow button from the organization switcher at the top of the page and select the organization from the dropdown. Once you've selected your organization, go to the **Integrations** tab to manage the integrations for the organization.
+- To configure a GitLab integration for your **Tenant**, click the double arrow button from the tenant switcher at the top of the Pipes console, select your tenant, and then select **Tenant Settings**. This option will only be visible in a custom tenant for which you are a [tenant owner](/pipes/docs/accounts/tenant/people#tenant-roles).  Once you've selected your tenant, go to the **Integrations** tab to manage the integrations for the tenant.
+
+Next, click the **New Integration** button. You will be asked to select the type of integration to be created.
+
+![](/images/docs/pipes/integrations_new_gitlab.png)
+
+Select **GitLab** and click **Next**.
+
+GitLab Integration configuration management form shows up. Enter the following information:
+- **Handle** for the integration. This handle should be meaningful and must be unique for all integrations in the tenant (including any org-level integrations).
+- **Host** for your GitLab installation which defaults to **gitlab.com**.
+- **Token** to be used by Pipes to discover projects. Please note that the token must be granted `api` access which is required to create and manage webhooks that enables Pipes to keep the mods managed by the integration up-to-date. Pipes supports [user](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html), [group](https://docs.gitlab.com/ee/user/group/settings/group_access_tokens.html) or [project](https://docs.gitlab.com/ee/user/project/settings/project_access_tokens.html) access tokens.
+- **Token Scope** is an optional section which allows you to further scope a token down to a GitLab group / project. The following cases might require you to configure this section:
+    - Using a **Personal Access Token** you need to mention the group / project to be used by Pipes to setup the webhook against.
+    - Using a **Group Access Token** and you want to further scope the webhook creation down to a group / project and restrict list of projects accessible to Pipes.
```

Comment:
Is this a single project?   this is confusing. it sounds like it can be a list of projects?? `scope the webhook creation down to a group / project and restrict list of projects accessible to Pipes.`.  Also, it suggests that this can be a group, but isn't that what the 'group scope' is for?

Reply:
I added a single `Token Scope` section similar to the UI, which we are using to further scope a token down to a group / project. The attributes that need to be set are indeed `group` / `project` and not `token scope`. 
The pointers outlines the use cases where you might need to use the token scope section as a whole. 
I'll be more explicit in what attributes one needs to set in each case to make it less confusing.

Reply:
Valid token type / filter combos are:

- Personal access token = you _must_ also provide a `group` or `project` path, which we will create the webhook at. It will also provide the "root" that we will search for projects within.
- Group access token = you _may_ optionally provide a `group` or `project` path, at which level which we will create the webhook at. It will also provide the "root" within the group that the access token belongs to where we will search for projects within.
- Project access token = you _may_ not provide any `group` or `project` path, as the token is already bound to a project, which if a lead node in the GitLab hierarchy.

Reply:
@MichaelBurgess 
- In the Group and Personal access token cases - is that `or` exclusive or inclusive?  Can you provide **both** a `group` and `project` path?
- Can we tell from the token structure which type it is?  Should we consider customizing the rest of the screen based on the token type?

Reply:
@sidr0cker would need to confirm - I believe it is exclusive and you can provide one, but not the other.